### PR TITLE
En_Dodojr (Baby Dodongo) OK

### PR DESCRIPTION
**Pull request by Zelllll**
Not much documentation this time, but that can always happen later.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c [hunk 1]
```diff
@@ -28,79 +50,591 @@ const ActorInit En_Dodojr_InitVars = {
...
+s32 func_809F68B0(EnDodojr* this, GlobalContext* globalCtx) {
+    if (this->actor.velocity.y >= 0.0f) {
+        return 0;
+    }
+
+    if (this->unk_1FC == 0) {
+        return 0;
+    }
+
+    if ((this->actor.bgCheckFlags & 1) != 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c:140]:
```suggestion
    if (this->actor.bgCheckFlags & 1) {
```

File: src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c [hunk 2]
```diff
@@ -28,79 +50,591 @@ const ActorInit En_Dodojr_InitVars = {
...
+
+void func_809F6E54(EnDodojr* this, GlobalContext* globalCtx) {
+    f32 angles[] = { 0.0f, 210.0f, 60.0f, 270.0f, 120.0f, 330.0f, 180.0f, 30.0f, 240.0f, 90.0f, 300.0f, 150.0f };
+    s32 pad;
+    Player* player = PLAYER;
+    Vec3f pos;
+    s16 angleIndex;
+
+    if (((this->bomb == NULL) || (this->bomb->update == NULL)) ||
+        (((this->bomb != NULL) && (this->bomb->parent != NULL)))) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c:278]:
```suggestion
    if ((this->bomb == NULL) || (this->bomb->update == NULL) ||
        ((this->bomb != NULL) && (this->bomb->parent != NULL))) {
```

File: src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c [hunk 3]
```diff
@@ -28,79 +50,591 @@ const ActorInit En_Dodojr_InitVars = {
...
+    EnBom* bomb;
+
+    if (this->unk_1FC != 0) {
+        if (this->actor.dmgEffectTimer == 0) {
+            func_8003426C(&this->actor, 0x4000, 200, 0, this->unk_1FC);
+            this->unk_1FC--;
+        }
+    } else {
+        bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.posRot.pos.x,
+                                   this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dodojr/z_en_dodojr.c:574]:
```suggestion
                                   this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, BOMB_BODY);
```

**Comment by fig02** [comment]
this needs conflicts resolved and actor/collision changes updated, but after that its good to go

**Review by Roman971 — APPROVED** [review]

**Comment by Zelllll** [comment]
Alright it should be all fixed now, and all of Roman's feedback has been applied

**Pull request merged**